spec/services/preserved_object_handler_results_spec.rb

```diff
@@ -69,14 +130,29 @@
   end
 
   context '#remove_db_updated_results' do
-    it 'needs tests' do
-      skip
+    before do
+      code = PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH
+      result_msg_args = { pc_version: 1, po_version: 2 }
+      pohr.add_result(code, result_msg_args)
+      code = PreservedObjectHandlerResults::PC_STATUS_CHANGED
+      result_msg_args = { old_status: PreservedCopy::OK_STATUS, new_status: PreservedCopy::INVALID_MOAB_STATUS }
+      pohr.add_result(code, result_msg_args)
+      code = PreservedObjectHandlerResults::CREATED_NEW_OBJECT
+      pohr.add_result(code)
+      code = PreservedObjectHandlerResults::INVALID_MOAB
+      pohr.add_result(code, 'foo')
     end
-  end
-
-  context '#result_hash' do
-    it 'needs tests' do
-      skip
+    it 'removes results matching DB_UPDATED_CODES' do
+      expect(pohr.result_array.size).to eq 4
+      pohr.remove_db_updated_results
+      expect(pohr.result_array.size).to eq 2
+      expect(pohr.result_array).not_to include(a_hash_including(PreservedObjectHandlerResults::CREATED_NEW_OBJECT))
+      expect(pohr.result_array).not_to include(a_hash_including(PreservedObjectHandlerResults::PC_STATUS_CHANGED))
+    end
+    it 'keeps results not matching DB_UPDATED_CODES' do
+      pohr.remove_db_updated_results
+      expect(pohr.result_array).to include(a_hash_including(PreservedObjectHandlerResults::PC_PO_VERSION_MISMATCH))
+      expect(pohr.result_array).to include(a_hash_including(PreservedObjectHandlerResults::INVALID_MOAB))
     end
```

Comment:
I'm happy with these tests, but one thought: it would be nice to be able to couple the expectations to `DB_UPDATED_CODES`. If I understand correctly, this will become easier once we do the work for #428.